core/object/object.cpp

```diff
@@ -808,6 +808,11 @@ String Object::to_string() {
 	return "[" + get_class() + ":" + itos(get_instance_id()) + "]";
 }
 
+void Object::get_undo_property_list(const StringName &p_property, const Variant &p_old_value, const Variant &p_new_value, List<PropertyChange> &r_property_list) const {
```

Comment:
I would probably just return  `List<StringName>` since you can query the value yourself

Reply:
Well, there are several problems with this approach, caused by the fact that I need to be able to access sub-properties of resources, while still being able to determine the owning object, as this is needed by the `add_undo_property(...)` function.

With a StringName, I cannot not access a sub-resource's properties (for example I could not use `get("texture.size")` in a Sprite2D node). So instead I though about using NodePath, which allow accessing sub-resources properties with a single string, but I have no way in such case to retrieve the Object owning the property, which is still needed by the `add_undo_property(...)` function.

If we go the NodePath route, I would likely need to add a new function, to retrieve the object owning a property. Right now we only have get_indexed(...) in Object that return the value, but no way to get the owning object. Using my previous example, `my_sprite_2D.get_indexed(..."texture.size"...)` returns the size value, and a new function `my_sprite_2D.get_indexed_object(..."texture.size"...)` would return the texture resource. I might also need a function to extract the property path for the whole path, like `my_sprite_2D.get_indexed_object(..."texture.size"...)` would return `"size"`.

That way I could solve the problem, but that's kind of more changes than what I propose here.

Reply:
ok, but why do you even need that?